book/P3C14_forests.qmd

```diff
@@ -0,0 +1,341 @@
+@fig-surv-ranfor-bootstrap demonstrates this process for three decision trees (panel a), where the survival probability is calculated at all possible time points (panels b-c), and the average is computed with linear interpolation added between time-points (panel d).
+
+![Bootstrapping Kaplan-Meier estimators across three decision trees (red, blue, green). Panel a) shows the individual estimates, b) shows the time points to aggregate the trees over, c) is the predicted survival probability from each tree at the desired time points, and d) is the average survival probabilities connected by a step function.](Figures/forests/bootstrap.png){#fig-surv-ranfor-bootstrap fig-alt="Four panels with 't' on x-axis and 'S(t)' on y-axis. Panel a) shows three survival functions as piece-wise linearly decreasing step functions. Panel (b) shows the same but with vertical dotted lines added at all time-points. Panel (c) shows dots at each of the intersections between the vertical lines and the plotted decision trees. Panel (d) shows a single line with dots which are the average of the points in panel (c)."}
+
+Extensions to competing risks follow naturally using bootstrapped cause-specific cumulative incidence functions.
+
+<!-- FIXME - ANDREAS TO ADD DETAILS ON COMPETING RISKS -->
+
+
+
+#### Deterministic Predictions {.unnumbered .unlisted}
+
+As discussed in @sec-eval-det, predicting and evaluating survival times is a complex and fairly under-researched area.
+For RSFs, there is an inclination to estimate survival times based on the mean or median survival times of observations in terminal nodes, however this would lead to biased estimations.
+Therefore, research has tended to focus on relative risk predictions.
+
+As discussed, relative risks are arbitrary values where only the resulting rank comparing observations matters.
+In RSFs, each terminal node should be as homogeneous as possible, hence within a terminal node, the risk between observation should be the same.
+Hence, one could theoretically assign a constant risk value to all terminal nodes, compare these using some concordant type metric, and shuffle repeatedly until maximum concordance is achieved.
+However, this would clearly be cumbersome and sub-optimal.
+Another approach would be to generate some heuristics based on the observations in the terminal nodes, for example the total number of events occurring in the terminal node, or the total number of events divided by the total time at risk, which gives an overall probability of the event occurring over time.
+
+In practice, the most common method appears to be a composition from the Nelson-Aalen method [@Ishwaran2008], which exploits results from counting process to provide a measure of expected mortality [@dataapplied].
+The relative risk predicted from decision tree $b$ is calculated as
+
+$$
+\phi_b := \sum_{\tau \in \calT} \hat{H}_b(\tau)
+$$
+
+where $\calT$ is the set of unique time-points in the leaf.
```

Comment:
which leaf? I think this should be \sum_h \hat{H}_{b(h)}, where the latter is from (14.3)?